src/Microsoft.AspNet.OData/Adapters/WebApiActionDescriptor.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Web.Http.Controllers;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Interfaces;
+
+namespace Microsoft.AspNet.OData.Adapters
+{
+    /// <summary>
+    /// Adapter class to convert Asp.Net WebApi action description to OData WebApi.
+    /// </summary>
+    internal class WebApiActionDescriptor : IWebApiActionDescriptor
+    {
+        /// <summary>
+        /// Gets the collection of supported HTTP methods for the descriptor.
+        /// </summary>
+        private IList<ODataRequestMethod> supportedHttpMethods;
+
+        /// <summary>
+        /// The inner action wrapped by this instance.
+        /// </summary>
+        private HttpActionDescriptor innerDescriptor;
+
+        /// <summary>
```

Comment:
Put <exception>...</exception> in the summaries? There are several constructors that require non-null parameters.

Reply:
I don't like "Won't Fix"ing comments but I am going to won't fix this. The null check pattern is used heavily in all OData code without the exception tags. I like the ide and it is proper but it's beyond the scope of this feature to add the tags everywhere that they are needed.